src_c/surface.c

```diff
@@ -2533,25 +2535,34 @@ surf_get_height(PyObject *self, PyObject *_null)
 }
 
 static PyObject *
-surf_get_rect(PyObject *self, PyObject *args, PyObject *kwargs)
+surf_generic_get_rect(PyObject *self, PyObject *const *args, Py_ssize_t nargs,
+                      PyObject *kwnames, char *type)
 {
-    PyObject *rect;
+    PyObject *rect = NULL;
     SDL_Surface *surf = pgSurface_AsSurface(self);
 
-    if (PyTuple_GET_SIZE(args) > 0) {
-        return RAISE(PyExc_TypeError,
-                     "get_rect only accepts keyword arguments");
+    if (nargs > 0) {
+        return PyErr_Format(PyExc_TypeError,
+                            "get_%s only accepts keyword arguments", type);
     }
 
     SURF_INIT_CHECK(surf)
 
-    rect = pgRect_New4(0, 0, surf->w, surf->h);
-    if (rect && kwargs) {
-        PyObject *key, *value;
-        Py_ssize_t pos = 0;
+    if (strcmp(type, "rect") == 0) {
+        rect = pgRect_New4(0, 0, surf->w, surf->h);
+    }
+    else if (strcmp(type, "frect") == 0) {
+        rect = pgFRect_New4(0.f, 0.f, (float)surf->w, (float)surf->h);
+    }
 
-        while (PyDict_Next(kwargs, &pos, &key, &value)) {
-            if ((PyObject_SetAttr(rect, key, value) == -1)) {
+    if (rect && kwnames) {
+        Py_ssize_t i, sequence_len;
+        PyObject **sequence_items;
+        sequence_items = PySequence_Fast_ITEMS(kwnames);
+        sequence_len = PySequence_Fast_GET_SIZE(kwnames);
```

Comment:
This can just be PyTuple_GET_SIZE.
```suggestion
        sequence_len = PyTuple_GET_SIZE(kwnames);
```